### ISSUE TITLE
Failure when running Roberta-base-11 model with dimension 1x1

### ISSUE THREAD

**Issue by hamptonm1** [onnx/onnx-mlir#2443]
Previously, we had an issues when compiling the Roberta model and the "work-around" was to set the pattern-shape-inference to false. However, the functionality has recently changed for pattern-shape-inference and it makes sense to get to the root of the problem. The new error I am seeing is as follows:

```
meganhampton@Megans-MBP-2 build % ./Debug/bin/onnx-mlir --O3 --EmitLib --maccel=NNPA roberta-base-11.onnx --onnx-op-stats TXT --shapeInformation 0:1x1 -v
The ONNX model has 124645716 elements in its initializers. This value would be close to and greater than the number of parameters in the model. Because there is no way to exactly count the number of parameters, this value can be used to have a rough idea of the number of parameters in the model.
Operations encountered:
-----------------------
   func.func          , 1
   func.return        , 1
   onnx.Add           , 58
   onnx.Cast          , 2
   onnx.Concat        , 77
   onnx.Constant      , 224
   onnx.CumSum        , 1
   onnx.Dim           , 116
   onnx.Div           , 41
   onnx.Equal         , 1
   onnx.Erf           , 12
   onnx.Gather        , 3
   onnx.Max           , 8
   onnx.Mul           , 32
   onnx.NoValue       , 1
   onnx.Not           , 1
   onnx.ReduceMeanV13 , 50
   onnx.Reshape       , 139
   onnx.Sqrt          , 25
   onnx.Sub           , 25
   onnx.Transpose     , 48
  zhigh.Add           , 73
  zhigh.Div           , 8
  zhigh.MatMul        , 97
  zhigh.Mul           , 43
  zhigh.Softmax       , 12
  zhigh.Stick         , 317
  zhigh.Tanh          , 1
  zhigh.Unstick       , 149
Assertion failed: (sizeInterface.isDynamicSize(unsignedIndex) && "Expected dynamic subview size"), function fold, file MemRefOps.cpp, line 1106.
zsh: abort      ./Debug/bin/onnx-mlir --O3 --EmitLib --maccel=NNPA roberta-base-11.onnx  TXT 
```

It seems like for dimensions that are `1x?` are all failing with this error (1x1, 1x2, 1x3 and so forth). Yet, if I use a `2x2` or anything else it works just fine.

**Comment by hamptonm1** [comment]
FYI- @sorenlassen @AlexandreEichenberger @tungld

**Comment by hamptonm1** [comment]
Here associated issues that were previously opened: https://github.com/onnx/onnx-mlir/pull/2270 and https://github.com/onnx/onnx-mlir/pull/2301

**Comment by hamptonm1** [comment]
Also I tried to debug by doing the following:

`meganhampton@Megans-MBP-2 build % ./Debug/bin/onnx-mlir --O3 --EmitZLowIR --maccel=NNPA roberta-base-11.onnx --onnx-op-stats TXT --shapeInformation 0:1x1 -v`

But no `roberta-base-11.onnx.mlir` file is generated which I was going to utilize to debug and figure out which pass is failing. Seems like it keeps hitting the error in MemRefOps.cpp, line 1106 before we can get any further.

**Comment by sorenlassen** [comment]
I tried it and I see it fails with
```
Assertion failed: (sizeInterface.isDynamicSize(unsignedIndex) && "Expected dynamic subview size"), function fold, file MemRefOps.cpp, line 1106.
```
which happens in FrontendToKrnlLoweringPass in ZHighStickOpShapeHelper::computeShape() in this call:
```
createIE->getShapeAsDims(input, inputDims);
```
where stickOp is
```
%8330 = "zhigh.Stick"(%8288) {layout = "3DS"} : (tensor<?x?x64xf32>) -> tensor<?x?x64xf32, #zhigh.layout<{dataLayout = "3DS"}>>
```

I only work in the ONNX dialect layer so please work with the ZHigh and Krnl experts first, but let me know if you trace back the problems to ONNX shape inference, then I can take a look

**Comment by tungld** [comment]
This error happens when lowering ONNX to Krnl, not related to accelerators. I tried to fix it by this PR #2446

**Comment by tungld** [comment]
This is a small example to reproduce the issue:
```mlir
func.func @test(%arg0 : tensor<12x5x5xf32>, %arg1: tensor<f32>) -> tensor<?x12x5x5xf32>{
  %35 = onnx.Constant dense<8.000000e+00> : tensor<f32>
  %0 = onnx.Constant dense<[1, 12, 5, 5]> : tensor<4xi64>
  %641 = "onnx.Reshape"(%arg0, %0) {allowzero = 0 : si64} : (tensor<12x5x5xf32>, tensor<4xi64>) -> tensor<?x12x5x5xf32>
  %1 = "onnx.Div"(%641, %35) {onnx_node_name = "Div_560"} : (tensor<?x12x5x5xf32>, tensor<f32>) -> tensor<?x12x5x5xf32>
  return %1 : tensor<?x12x5x5xf32>
}
```

We will see the issue if lowering it to krnl by using `Debug/bin/onnx-mlir-opt --convert-onnx-to-krnl test.mlir`.

There is no problem if we do `--shape-inference` first: `Debug/bin/onnx-mlir-opt --shape-inference --convert-onnx-to-krnl test.mlir`

**Comment by hamptonm1** [comment]
@tungld I am still seeing the above error after pulling in your changes and rebuilding. Can you do a quick test locally to confirm if you see the issue resolved on your end? I want to make sure I am not going  crazy here lol

**Comment by hamptonm1** [comment]
Okay @tungld new PR solves this problem :). I will close out the issue once we merge in the latest code. Thanks!

**Comment by hamptonm1** [comment]
I merged the PR! So we can close this one out for now :)
